### PR TITLE
Spipe cleanup and valgrind

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: spipe/main.c [hunk 1]
```diff
@@ -212,6 +214,16 @@ main(int argc, char * argv[])
...
+	/* Wait for threads to finish. */
+	if ((rc = pthread_join(threads[0], NULL)) != 0) {
+		warn0("pthread_join: %s", strerror(rc));
+		goto err2;
+	}
+	if ((rc = pthread_join(threads[1], NULL)) != 0) {
+		warn0("pthread_join: %s", strerror(rc));
+		goto err2;
+	}
+
```
Comment (cperciva) [inline spipe/main.c]: These will end up calling `sock_addr_freelist(sas_t);` if they fail.  Is that correct?

**Comment by gperciva** [comment]
This now depends on #224 and #226.

**Comment by gperciva** [comment]
@TimWolla reminded us that this is similar to #44.  In particular, @cperciva had some concerns about what if we're writing to a process which stalled https://github.com/Tarsnap/spiped/pull/44#issuecomment-210781703.

**Comment by gperciva** [comment]
@TimWolla reminded us that this is similar to #44.  In particular, @cperciva had some concerns about what if we're writing to a process which stalled https://github.com/Tarsnap/spiped/pull/44#issuecomment-210781703.

**Comment by gperciva** [comment]
@TimWolla reminded us that this is similar to #44.  In particular, @cperciva had some concerns about what if we're writing to a process which stalled https://github.com/Tarsnap/spiped/pull/44#issuecomment-210781703.

File: spipe/main.c (hunk 1 above)
Comment (gperciva) [reply spipe/main.c]: fixed in #224.

Comment (gperciva) [reply spipe/main.c]: Or rather, fix *proposed* in #224.

Comment (gperciva) [reply spipe/main.c]: Or rather, fix *proposed* in #224.

**Comment by gperciva** [comment]
Ready for merge.

**Pull request merged**